p2p/test/muxer-negotiation/muxer_negotiation_test.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/p2p/muxer/mplex"
+	"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	"github.com/stretchr/testify/require"
+
+	/*
+		"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	*/
+
+	golog "github.com/ipfs/go-log/v2"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const testData = "Muxer Test Data\n"
+
+type muxerTest struct {
+	svrMuxers     []string
+	svrTrans      []network.Multiplexer
+	cliMuxers     []string
+	cliTrans      []network.Multiplexer
+	expectedMuxer string
+}
+
+func TestMuxerNegotiatin(t *testing.T) {
+	secTypes := []string{Tls, Noise}
+	testCases := []muxerTest{
+		{svrMuxers: []string{"/mplex/6.7.0"},
+			svrTrans:      []network.Multiplexer{mplex.DefaultTransport},
+			cliMuxers:     []string{"/yamux/1.0.0"},
+			cliTrans:      []network.Multiplexer{yamux.DefaultTransport},
+			expectedMuxer: ""},
+		{svrMuxers: []string{"/mplex/6.7.0"},
+			svrTrans:      []network.Multiplexer{mplex.DefaultTransport},
+			cliMuxers:     []string{"/mplex/6.7.0"},
+			cliTrans:      []network.Multiplexer{mplex.DefaultTransport},
+			expectedMuxer: "/mplex/6.7.0"},
+		{svrMuxers: []string{"/yamux/1.0.0"},
+			svrTrans:      []network.Multiplexer{yamux.DefaultTransport},
+			cliMuxers:     []string{"/yamux/1.0.0"},
+			cliTrans:      []network.Multiplexer{yamux.DefaultTransport},
+			expectedMuxer: "/yamux/1.0.0"},
+		{svrMuxers: []string{"/yamux/1.0.0", "/mplex/6.7.0"},
+			svrTrans:      []network.Multiplexer{yamux.DefaultTransport, mplex.DefaultTransport},
+			cliMuxers:     []string{"/mplex/6.7.0", "/yamux/1.0.0"},
+			cliTrans:      []network.Multiplexer{mplex.DefaultTransport, yamux.DefaultTransport},
+			expectedMuxer: "/yamux/1.0.0"},
+		{svrMuxers: []string{"/mplex/6.7.0", "/yamux/1.0.0"},
+			svrTrans:      []network.Multiplexer{mplex.DefaultTransport, yamux.DefaultTransport},
+			cliMuxers:     []string{"/yamux/1.0.0", "/mplex/6.7.0"},
+			cliTrans:      []network.Multiplexer{yamux.DefaultTransport, mplex.DefaultTransport},
+			expectedMuxer: "/mplex/6.7.0"},
+	}
+
+	doMuxerNegotiation := func(t *testing.T, secType string, svrMuxers, cliMuxers []string, svrTrans, cliTrans []network.Multiplexer, expected string) {
+		sctx, sCancel := context.WithCancel(context.Background())
+		cctx, cCancel := context.WithCancel(context.Background())
+
+		sSec, svrh, err := makeHost(secType, svrMuxers, svrTrans, 58568)
+		require.NoError(t, err)
+		require.NotNil(t, sSec)
+		require.NotNil(t, svrh)
+
+		cSec, clih, err := makeHost(secType, cliMuxers, cliTrans, 59569)
+		require.NoError(t, err)
+		require.NotNil(t, cSec)
+		require.NotNil(t, clih)
+
+		ready := make(chan struct{})
+		go func() {
+			runServer(sctx, svrh)
+			close(ready)
+			<-sctx.Done()
+		}()
+
+		<-ready
+		runClientAndCheckMuxer(cctx, clih, getHostAddress(svrh), cSec, sSec, expected, t)
+		clih.Close()
+		svrh.Close()
+		cCancel()
+		sCancel()
+	}
+
+	golog.SetAllLoggers(golog.LevelInfo)
```

Comment:
Why is this needed?

Reply:
Not needed anymore, used for debugging. thanks for catching it